src/extension.ts

```diff
@@ -73,6 +75,29 @@ export async function activate(context: ExtensionContext) {
       }
     });
   });
+
+  if (!hasShownQuteValidationPopUp()) {
+    await showQuteValidationPopUp();
+  }
+
+}
+
+function hasShownQuteValidationPopUp(): boolean {
+  return QuarkusContext.getExtensionContext().globalState.get(EXPERIMENTAL_QUTE_VALIDATION_FLAG, false);
+}
+
+async function showQuteValidationPopUp() {
+  const EXPERIMENTAL_QUTE_VALIDATION_ADVERTISEMENT = 'Enable experimental validation for Qute files?' + //
+        '(You may change this setting later in `package.json`)';
+  const ENABLE_MESSAGE = `Enable`;
+  const DONT_SHOW_AGAIN_MESSAGE = "Don't show this again";
+  const input = await window.showInformationMessage(EXPERIMENTAL_QUTE_VALIDATION_ADVERTISEMENT, ENABLE_MESSAGE, DONT_SHOW_AGAIN_MESSAGE);
+  if (input === ENABLE_MESSAGE) {
+    workspace.getConfiguration('settings', workspace.workspaceFolders[0].uri).update("quarkus.tools.qute.validation.enabled", true)
```

Comment:
I will create a `settings.ts` file in https://github.com/redhat-developer/vscode-quarkus/tree/master/src/qute

```ts
/**
 * VScode Qute settings.
 */
export namespace QuteSettingss {

  /**
   * Enable/disable all Qute validation settings.
   */
  export const QUTE_VALIDATION_ENABLED = quarkus.tools.qute.validation.enabled';

}
```
to write

```ts
workspace.getConfiguration('settings', workspace.workspaceFolders[0].uri).update(QuteSettings.QUTE_VALIDATION_ENABLED, true)
```

Reply:
see how to get the proper workspace from the file uri : https://github.com/microsoft/vscode/wiki/Adopting-Multi-Root-Workspace-APIs#eliminating-rootpath